fix: Use Assembly value as identifier, not name

zealws opened the pull request:
I noticed an issue in FUSOR where the translator was using the incorrect genomic version.

```
SeqRepo unable to get translated identifiers for Assembly.GRCH38:chr1
Genomic accession for chr1 could not be retrieved
ValueError encountered while loading records from jaffa_sample.txt
Traceback (most recent call last):
  File "/Users/zxw016/dev/wagner/fusor/src/fusor/harvester.py", line 116, in load_records
    translated_fusion = await self.translator.translate(
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        fusion, self.coordinate_type, self.assembly
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "/Users/zxw016/dev/wagner/fusor/src/fusor/translator.py", line 319, in translate
    genomic_ac=self._get_genomic_ac(jaffa.chrom1, rb),
               ~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^
  File "/Users/zxw016/dev/wagner/fusor/src/fusor/translator.py", line 217, in _get_genomic_ac
    raise ValueError
ValueError
1 fusion(s) were dropped during translation
```

You can see in the first line of the error, it's attempting to query seqrepo using `Assembly.GRCH38:chr1`, rather than `GRCh38:chr1`.

The [translator formats that value](https://github.com/cancervariants/fusor/blob/main/src/fusor/translator.py#L212) using a `str` call on the Assembly (implicit via an f-string). Because this value is an enum, the result is the enum's key. The `value` field has the correctly formatted string:

```
>>> a = Assembly.GRCH38
>>> str(a)
'Assembly.GRCH38'
>>> a.value
'GRCh38'
```

I modified the translator to call `value` on the Assembly. I also fixed the unit tests which were passing in a `str` value as build.

<details>
<summary>Output before:</summary>

```
> ./test-fusor.py jaffa                                                                                                                
SeqRepo unable to get translated identifiers for Assembly.GRCH38:chr1
Genomic accession for chr1 could not be retrieved
ValueError encountered while loading records from jaffa_sample.txt
Traceback (most recent call last):
  File "/Users/zxw016/dev/wagner/fusor/src/fusor/harvester.py", line 116, in load_records
    translated_fusion = await self.translator.translate(
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        fusion, self.coordinate_type, self.assembly
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "/Users/zxw016/dev/wagner/fusor/src/fusor/translator.py", line 319, in translate
    genomic_ac=self._get_genomic_ac(jaffa.chrom1, rb),
               ~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^
  File "/Users/zxw016/dev/wagner/fusor/src/fusor/translator.py", line 217, in _get_genomic_ac
    raise ValueError
ValueError
1 fusion(s) were dropped during translation
```
</details>

<details>
<summary>Output after:</summary>

```
>  ./test-fusor.py jaffa                                                                                                                
{"type":"AssayedFusion","regulatoryElement":null,"structure":[{"type":"TranscriptSegmentElement","transcript":"refseq:NM_170707.4","strand":1,"exonStart":null,"exonStartOffset":null,"exonEnd":2,"exonEndOffset":0,"gene":{"id":null,"extensions":null,"conceptType":"Gene","name":"LMNA","primaryCoding":{"id":"hgnc:6636","extensions":null,"name":null,"system":"https://www.genenames.org/data/gene-symbol-report/#!/hgnc_id/","systemVersion":null,"code":"HGNC:6636","iris":null},"mappings":null},"elementGenomicStart":null,"elementGenomicEnd":{"id":"ga4gh:SL.99a9r3W9FBpCbCoMKpypLNDzjICf9k6X","type":"SequenceLocation","name":null,"description":null,"aliases":null,"extensions":null,"digest":"99a9r3W9FBpCbCoMKpypLNDzjICf9k6X","sequenceReference":{"id":"refseq:NC_000001.11","type":"SequenceReference","name":null,"description":null,"aliases":null,"extensions":null,"refgetAccession":"SQ.Ya6Rs7DHhDeg7YaOSg1EoNi3U_nQ9SvO","residueAlphabet":null,"circular":null,"sequence":null,"moleculeType":null},"start":null,"end":156130773,"sequence":null},"coverage":null,"anchoredReads":null},{"type":"TranscriptSegmentElement","transcript":"refseq:NM_002529.4","strand":1,"exonStart":11,"exonStartOffset":0,"exonEnd":null,"exonEndOffset":null,"gene":{"id":null,"extensions":null,"conceptType":"Gene","name":"NTRK1","primaryCoding":{"id":"hgnc:8031","extensions":null,"name":null,"system":"https://www.genenames.org/data/gene-symbol-report/#!/hgnc_id/","systemVersion":null,"code":"HGNC:8031","iris":null},"mappings":null},"elementGenomicStart":{"id":"ga4gh:SL.w6FL8tvp9-KPUBj19HufOFJVmu9CKnyh","type":"SequenceLocation","name":null,"description":null,"aliases":null,"extensions":null,"digest":"w6FL8tvp9-KPUBj19HufOFJVmu9CKnyh","sequenceReference":{"id":"refseq:NC_000001.11","type":"SequenceReference","name":null,"description":null,"aliases":null,"extensions":null,"refgetAccession":"SQ.Ya6Rs7DHhDeg7YaOSg1EoNi3U_nQ9SvO","residueAlphabet":null,"circular":null,"sequence":null,"moleculeType":null},"start":156874905,"end":null,"sequence":null},"elementGenomicEnd":null,"coverage":null,"anchoredReads":null}],"readingFramePreserved":null,"viccNomenclature":"NM_170707.4(LMNA):e.2::NM_002529.4(NTRK1):e.11","causativeEvent":null,"assay":null,"contig":null,"readData":{"type":"ReadData","split":{"type":"SplitReads","splitReads":6},"spanning":{"type":"SpanningReads","spanningReads":3}}}
```
</details>


<details>
<summary>Pytest results</summary>

```
> pytest
========================================================================= test session starts ==========================================================================
platform darwin -- Python 3.13.5, pytest-8.4.1, pluggy-1.6.0
rootdir: /Users/zxw016/dev/wagner/fusor
configfile: pyproject.toml
testpaths: tests
plugins: anyio-4.9.0, asyncio-1.1.0, cov-6.2.1
asyncio: mode=Mode.AUTO, asyncio_default_fixture_loop_scope=function, asyncio_default_test_loop_scope=function
collected 55 items                                                                                                                                                     

tests/test_fusion_matching.py .                                                                                                                                  [  1%]
tests/test_fusor.py ..........                                                                                                                                   [ 20%]
tests/test_harvesters.py .........                                                                                                                               [ 36%]
tests/test_models.py .....................                                                                                                                       [ 74%]
tests/test_nomenclature.py ..                                                                                                                                    [ 78%]
tests/test_tools.py ..                                                                                                                                           [ 81%]
tests/test_translators.py ..........                                                                                                                             [100%]

============================================================================ tests coverage ============================================================================
___________________________________________________________ coverage: platform darwin, python 3.13.5-final-0 ___________________________________________________________

Name                                Stmts   Miss  Cover   Missing
-----------------------------------------------------------------
src/fusor/__init__.py                   8      2    75%   12-13
src/fusor/config.py                    12      1    92%   25
src/fusor/examples/__init__.py         24      0   100%
src/fusor/exceptions.py                 2      0   100%
src/fusor/fusion_caller_models.py     115      0   100%
src/fusor/fusion_matching.py           98     16    84%   42-43, 45, 66-67, 96, 183, 192, 232-238, 250, 280-281
src/fusor/fusor.py                    203     18    91%   75, 114-115, 123-124, 158-159, 275-280, 294, 463, 503, 512-515, 559
src/fusor/harvester.py                146      6    96%   85-86, 119-120, 292, 318
src/fusor/models.py                   301      7    98%   281-282, 585-586, 606, 685-686
src/fusor/nomenclature.py              94      9    90%   48-49, 55, 58, 115-116, 118-119, 172
src/fusor/tools.py                     42     17    60%   40, 73-92, 106
src/fusor/translator.py               295     44    85%   143, 172-181, 197-200, 215-217, 237, 273-275, 314, 390, 462, 531, 602, 730, 757, 796-829, 857, 920, 997, 1026, 1052
-----------------------------------------------------------------
TOTAL                                1340    120    91%
========================================================================= 55 passed in 50.85s ==========================================================================
</details>

jsstevenson commented on src/fusor/translator.py:
```diff
@@ -199,7 +199,7 @@ def _are_fusion_partners_different(
         return False
 
-    def _get_genomic_ac(self, chrom: str, build: Assembly) -> str:
+    def _get_genomic_ac(self, chrom: str, build: Assembly | str) -> str:
```
I don't know if I've ever looked at this code before, so take this with a grain of salt. But my perhaps arbitrary stylistic preference is that stricter type signatures are better, and that this type coercion should happen closer to wherever the `str` value is coming from in the first place. Just my two cents.

zealws replied:
That's fair, I was optimizing for a shorter PR, but fixing the tests to pass the enum value instead would probably be better in the long run.

jsstevenson replied:
Yup, totally reasonable. This wouldn't be the only place where we employ the `str | Enum` pattern. Feel free to ignore especially if the alternative is nontrivial

zealws commented:
I added 31e057e which changes all the unit-tests which were passing `SomeEnum.value` (a string) in place of `SomeEnum` (a value of the enum class). This include `Assembly` values, but also there were a lot of `CoordinateType` values being passed in the same way, so I corrected those as well.

`pytest` still passes, except for an error in `tests/test_fusion_matching.py` which fails in `main`, so I assume that's work in progress.

korikuzma commented:
@jsstevenson Did you want to review again or can I go ahead and merge?
